Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -228,6 +240,7 @@ FieldValue Serializer::DecodeFieldValue(const uint8_t* bytes, size_t length) {
       break;
 
     case google_firestore_v1beta1_Value_string_value_tag:
+    case google_firestore_v1beta1_Value_map_value_tag:
       if (wire_type != PB_WT_STRING) {
```

Comment:
This would imply a corrupt message; I'm not sure an assertion properly applies since we should be able to detect that and return an error code rather than just failing an assertion or aborting. Either way, I'm going to defer this to the error handling PR (to come shortly).